main.go

```diff
@@ -58,18 +67,11 @@ func readConfig() config {
 }
 
 func startHttpsServer(simpleServer httpd.SimpleServer) error {
-	errs := make(chan error, 1)
-	simpleServer.Start(errs)
-
-	var returnErr error
-	select {
-	case err := <-errs:
-		returnErr = err
-	case <-time.After(5 * time.Second):
-	}
-
-	close(errs)
-	return returnErr
+	errC := make(chan error, 1)
+	defer close(errC)
+	simpleServer.Start(errC)
+	err := <-errC
```

Comment:
This operation will block until you receive an error through the channel. `select` is not necessary as you don't do anything on the time.After.

Reply:
I didnt want to block forever. After starting the server, I want to return back to main where it waits for os.signal for SIGTERM